### ISSUE TITLE
Hover & text on shapes

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.js#4780]
We don't want layout shapes to be full traces but it would be pretty convenient to be able to add hoverlabels and text to them, for something like object-annotation or to label horizontal lines on plots.

**Comment by jorgenherje** [comment]
Would be nice with hover text for shapes. 

Shapes are nice to utilize for marking/annotation in plots with vertical/horizontal lines. However it would be nice to have hover text, s.t. info is not visible at all times.

Working in [WebViz](https://github.com/equinor/webviz-subsurface) using Dash and plotly. I've been testing some ways to obtain vertical lines with text info, and found the two possible solutions so far:

1) Shapes + annotations (text labels always visible - not optimal):  
![image](https://user-images.githubusercontent.com/82032112/150308595-2f28a1b3-03dc-46ff-be60-04c96efc848a.png)

2) Vertical trace (must add after data, and calculate min and max y-value. Can become slow with many traces? And does not span across y-axis when zooming out):
![image](https://user-images.githubusercontent.com/82032112/150308895-ae53c73f-3ac2-4d6a-b5b7-0328f58d414b.png)

**Comment by RichardNeill** [comment]
Agree, this would be great. If we can't have the full "hover" mechanism, what about enabling
the HTML title attribute?

**Comment by iacisme** [comment]
I'd like to bring this up as well. It'd be really cool if hover data could be added to vline or hline objects.

![image](https://github.com/plotly/plotly.js/assets/72644814/b70a9f7a-f143-4198-ad92-9ffc45cb4ebd)

**Comment by matanox** [comment]
I'm not aware there's any issue adding text on shapes these days, but hover labels/text for shapes would be nice.

**Comment by jaladh-singhal** [comment]
This will be super useful for charts that rely on shapes (like line, rectangle) as markers to see more info about a marker. In my case, I cannot add a hidden decorative scatter trace to emulate hover tip behavior.

Are there any plans to implement it in the near future? As of plotly.js v3.6, shapes already support label/text so I'm anticipating if it will come next (given that annotations also support hover tips)?

**Comment by robertclaus** [comment]
This would be a really interesting issue for an ambitious community member to open a PR for!
